src/query/service/src/pipelines/processors/transforms/transform_udf_server.rs

```diff
@@ -79,6 +78,20 @@ impl TransformUdfServer {
                 UDFFlightClient::build_endpoint(server_addr, connect_timeout, request_timeout)?;
             endpoints.insert(server_addr.clone(), endpoint);
         }
+        Ok(endpoints)
+    }
+
+    pub fn new(
+        ctx: Arc<QueryContext>,
+        funcs: Vec<UdfFunctionDesc>,
+        endpoints: BTreeMap<String, Arc<Endpoint>>,
+    ) -> Result<Self> {
+        let settings = ctx.get_settings();
+        let connect_timeout = settings.get_external_server_connect_timeout_secs()?;
+        let request_batch_rows = settings.get_external_server_request_batch_rows()? as usize;
+        let request_max_threads = settings.get_external_server_request_max_threads()? as usize;
+        let retry_times = settings.get_external_server_request_retry_times()? as usize;
+        let semaphore = Arc::new(Semaphore::new(request_max_threads));
```

Comment:
seems this is not query level but transform level.

Reply:
fixed